engine/runner/src/main/scala/org/enso/runner/Main.scala

```diff
@@ -756,7 +756,10 @@ object Main {
       val mainCons = mainModule.getAssociatedConstructor
       val mainFun  = mainModule.getMethod(mainCons, mainMethodName)
       mainFun match {
-        case Some(main) => main.execute()
+        case Some(main) if mainMethodName != "main" =>
+          main.execute(mainCons.newInstance())
```

Comment:
A bit hacky check, I'd say. Could we pass `mainCons.newInstance()` all the time? If not, shalln't we rather call without the instances and if that fails try again with the instance?

Reply:
Totally agree this sounds hacky. Implicit self is exceptionally not added to `main` in general (see https://github.com/enso-org/enso/blob/develop/engine/runtime/src/main/scala/org/enso/compiler/pass/desugar/GenerateMethodBodies.scala#L240) . This was done on purpose as it simplified a ton of logic all over the place. In that sense `main` is trully static.
I assumed that `main` will always be `main` but turned out not to be the case for REPL which for some reason happens to be named `internal_repl_entry_point___` :shrug: 

Once we truly have statics and implicit self is no longer added, this hack will go as well.